File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/horario/HorarioMesResponse.java
package pe.albrugroup.schedule_service.entity.response.horario;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class HorarioMesResponse {

    private Long idEmpleado;
    private Integer anio;
    private Integer mes;
    private List<HorarioMesVigenciaResponse> vigencias;
}

File: frontend/src/app/shared/models/schedule/horario-mes-response.ts
import { ExcepcionHorarioResponse } from './excepcion-horario-response';

export interface HorarioMesVigenciaResponse {
  idHorario: number;
  modalidad: string;
  horasObjetivoSemanal: number;
  horasObjetivoMensual: number;
  minutosAlmuerzo: number;
  minutosServicios: number;
  fechaInicio: string;
  fechaFin: string | null;
  desdeAplicacion: string;
  hastaAplicacion: string;
  compensable: boolean;
  detallesBase: Array<{
    id: number;
    dia: string;
    horaEntrada: string;
    horaSalida: string;
    inicioAlmuerzo: string | null;
    finAlmuerzo: string | null;
    laborable: boolean;
  }>;
  modificaciones: ExcepcionHorarioResponse[];
}

export interface HorarioMesResponse {
  idEmpleado: number;
  anio: number;
  mes: number;
  vigencias: HorarioMesVigenciaResponse[];
}
